Src/Plugins/Security/SEC103_017.PfxCryptographicKeyfileValidator.cs

```diff
@@ -11,29 +11,35 @@ namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
 {
     public static class PfxCryptographicKeyfileValidator
     {
-        public static ValidationState IsValidStatic(ref string matchedPattern,
-                                                    ref Dictionary<string, string> groups,
-                                                    ref string message,
-                                                    out ResultLevelKind resultLevelKind,
-                                                    out Fingerprint fingerprint)
+        public static IEnumerable<ValidationResult> IsValidStatic(ref string matchedPattern,
+                                                                  ref Dictionary<string, string> groups,
+                                                                  ref string message)
         {
-            fingerprint = default;
-            resultLevelKind = default;
-
             groups.TryGetValue("content", out string content);
 
             if (!string.IsNullOrWhiteSpace(content) &&
                 !content.Any(ch => char.IsControl(ch) && ch != '\r' && ch != '\n'))
             {
                 // This condition indicates that we have textual (PEM-encoded) data.
                 // These certificates are handled by the SecurePlaintextSecrets rules.
-                return ValidationState.NoMatch;
+                return ValidationResult.NoMatch;
             }
 
-            return CertificateHelper.TryLoadCertificate(matchedPattern,
-                                                        ref fingerprint,
-                                                        ref message,
-                                                        ref resultLevelKind);
+            Fingerprint fingerprint = default;
```

Comment:
>Fingerprint fingerprint = default; [](http://example.com/codeflow?start=12&length=34)

the initializations are no longer required and are just taking up space, can we remove them all?

Reply:
actually, we do need this. because we need the fingerprint and resultlevelkind to be populated in the following method.